supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/MachineConstraints.scala

```diff
@@ -8,10 +8,18 @@ import mouse.all._
 import org.slf4j.Logger
 import wdl4s.parser.MemoryUnit
 import wom.format.MemorySize
+import scala.math.BigDecimal.RoundingMode.CEILING
+
+case class TypeCastMismatch(message: String) extends Exception(message)
 
 object MachineConstraints {
   implicit class EnhancedInformation(val information: MemorySize) extends AnyVal {
-    def asMultipleOf(factor: MemorySize): MemorySize = MemorySize(factor.amount * (information.bytes / factor.bytes).ceil, factor.unit)
+    def ceil(a: BigDecimal): Double = {
+      if (a.isDecimalDouble) a.setScale(0, CEILING).doubleValue
+      else throw TypeCastMismatch("Can not allocate memory")
+    }
+
+    def asMultipleOf(factor: MemorySize): MemorySize = MemorySize(factor.amount * ceil(information.bytes / factor.bytes), factor.unit)
```

Comment:
Both `information.bytes` and `factor.bytes` are Double. Why are you ceiling it with a function that takes BigDecimal as an argument?